app/ai/itinerary_nodes/plan.py

```
@@ -0,0 +1,91 @@
+from .state import PlanState
+
+
+def node_plan_days(state: PlanState) -> PlanState:
+    poi_items = state.get("poi_candidates") or []
+    nights = state.get("nights")
+    if nights is None and state.get("date_range"):
+        dr = DateRange(**state["date_range"])  # type: ignore
+        nights = (dr.end - dr.start).days
+    try:
+        nights = int(nights) if nights is not None else None
+    except Exception:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)